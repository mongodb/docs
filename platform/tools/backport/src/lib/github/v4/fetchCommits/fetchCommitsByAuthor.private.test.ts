import { DocumentNode, print } from 'graphql';
import { getDevAccessToken } from '../../../../test/private/getDevAccessToken';
import { Commit } from '../../../sourceCommit/parseSourceCommit';
import * as apiRequestV4Module from '../apiRequestV4';
import { fetchCommitsByAuthor } from './fetchCommitsByAuthor';

const accessToken = getDevAccessToken();

describe('fetchCommitsByAuthor', () => {
  describe('snapshot request/response', () => {
    let spy: jest.SpyInstance;
    let commits: Commit[];

    beforeEach(async () => {
      spy = jest.spyOn(apiRequestV4Module, 'apiRequestV4');
      commits = await fetchCommitsByAuthor({
        accessToken,
        author: 'sorenlouv',
        maxNumber: 10,
        repoName: 'kibana',
        repoOwner: 'elastic',
        sourceBranch: 'main',
        dateSince: '2021-01-10T00:00:00Z',
        dateUntil: '2022-01-01T00:00:00Z',
        commitPaths: [] as Array<string>,
      });
    });

    it('makes the right queries', () => {
      const queries = spy.mock.calls.reduce((acc, call) => {
        const query = call[0].query as DocumentNode;
        //@ts-expect-error
        const name = query.definitions[0].name.value;
        return { ...acc, [name]: print(query) };
      }, {});

      const queryNames = Object.keys(queries);
      expect(queryNames).toEqual(['AuthorId', 'CommitsByAuthor']);

      queryNames.forEach((name) => {
        expect(queries[name]).toMatchSnapshot(`Query: ${name}`);
      });
    });

    it('returns the correct response', async () => {
      expect(commits).toMatchSnapshot();
    });
  });

  describe('commitPaths', () => {
    const getOptions = () => ({
      accessToken,
      author: 'sorenlouv',
      maxNumber: 10,
      repoName: 'repo-with-different-commit-paths',
      repoOwner: 'backport-org',
      sourceBranch: 'main',
      dateSince: null,
      dateUntil: null,
    });

    const getCommitMessages = (commits: Commit[]) => {
      return commits.map((c) =>
        c.sourceCommit.message.replace(/(\r\n|\n|\r)/gm, ''),
      );
    };

    it('returns all commits', async () => {
      const commits = await fetchCommitsByAuthor({
        ...getOptions(),
        commitPaths: [] as Array<string>,
      });

      const commitMessages = getCommitMessages(commits);
      expect(commitMessages).toEqual([
        'Edit all lyrics',
        'Update .backportrc.json',
        'Edit "Take on me"',
        'Add backportrc.json',
        'Add "Bohemian Rhapsody""',
        'Add "99 Luftballons"',
        'Add "Take on me"',
      ]);
    });

    it('only returns commits related to "99-luftballons.txt"', async () => {
      const commits = await fetchCommitsByAuthor({
        ...getOptions(),
        commitPaths: ['lyrics/99-luftballons.txt'],
      });

      const commitMessages = getCommitMessages(commits);
      expect(commitMessages).toEqual([
        'Edit all lyrics',
        'Add "99 Luftballons"',
      ]);
    });

    it('only returns commits related to "take-on-me.txt"', async () => {
      const commits = await fetchCommitsByAuthor({
        ...getOptions(),
        commitPaths: ['lyrics/take-on-me.txt'],
      });

      const commitMessages = getCommitMessages(commits);
      expect(commitMessages).toEqual([
        'Edit all lyrics',
        'Edit "Take on me"',
        'Add "Take on me"',
      ]);
    });

    it('removes duplicates and order by `committedDate`', async () => {
      const commits = await fetchCommitsByAuthor({
        ...getOptions(),
        commitPaths: [
          'lyrics/99-luftballons.txt',
          'lyrics/take-on-me.txt',
          'lyrics/bohemian-rhapsody.txt',
        ],
      });

      const commitMessages = getCommitMessages(commits);
      expect(commitMessages).toEqual([
        'Edit all lyrics',
        'Edit "Take on me"',
        'Add "Bohemian Rhapsody""',
        'Add "99 Luftballons"',
        'Add "Take on me"',
      ]);
    });
  });

  describe('targetPullRequestStates', () => {
    let res: Awaited<ReturnType<typeof fetchCommitsByAuthor>>;
    beforeEach(async () => {
      res = await fetchCommitsByAuthor({
        accessToken,
        commitPaths: [],
        maxNumber: 10,
        repoName: 'backport-e2e',
        repoOwner: 'backport-org',
        sourceBranch: 'master',
        author: null,
        dateSince: null,
        dateUntil: null,
      });
    });

    it('returns related OPEN PRs', async () => {
      const commitWithOpenPR = res.find(
        (commit) => commit.sourcePullRequest?.number === 9,
      );
      expect(commitWithOpenPR?.targetPullRequestStates).toEqual([
        {
          branch: '7.8',
          label: 'v7.8.0',
          branchLabelMappingKey: '^v(\\d+).(\\d+).\\d+$',
          isSourceBranch: false,
          state: 'OPEN',
          number: 10,
          url: 'https://github.com/backport-org/backport-e2e/pull/10',
        },
      ]);
    });

    it('returns related MERGED PRs', async () => {
      const commitWithMergedPRs = res.find(
        (commit) => commit.sourcePullRequest?.number === 5,
      );
      expect(commitWithMergedPRs?.targetPullRequestStates).toEqual([
        {
          branch: '7.8',
          label: 'v7.8.0',
          branchLabelMappingKey: '^v(\\d+).(\\d+).\\d+$',
          isSourceBranch: false,
          mergeCommit: {
            message: 'Add 🍏 emoji (#5) (#7)',
            sha: '46cd6f9999effdf894a36dbc7db90e890f4be840',
          },
          number: 7,
          state: 'MERGED',
          url: 'https://github.com/backport-org/backport-e2e/pull/7',
        },
        {
          branch: '7.x',
          label: 'v7.9.0',
          branchLabelMappingKey: '^v7.9.0$',
          isSourceBranch: false,
          mergeCommit: {
            message: 'Add 🍏 emoji (#5) (#6)',
            sha: '4bcd876d4ceaa73cf437bfc89b74d1a4e704c0a6',
          },
          number: 6,
          state: 'MERGED',
          url: 'https://github.com/backport-org/backport-e2e/pull/6',
        },
        {
          branch: 'master',
          label: 'v8.0.0',
          branchLabelMappingKey: '^v8.0.0$',
          isSourceBranch: true,
          mergeCommit: {
            message: 'Add 🍏 emoji (#5)',
            sha: 'ee8c492334cef1ca077a56addb79a26f79821d2f',
          },
          number: 5,
          state: 'MERGED',
          url: 'https://github.com/backport-org/backport-e2e/pull/5',
        },
      ]);
    });

    it('returns missing pull requests', async () => {
      const commitWithoutPRs = res.find(
        (commit) => commit.sourcePullRequest?.number === 8,
      );
      expect(commitWithoutPRs?.targetPullRequestStates).toEqual([
        {
          branch: '7.x',
          label: 'v7.9.0',
          branchLabelMappingKey: '^v7.9.0$',
          isSourceBranch: false,
          state: 'NOT_CREATED',
        },
        {
          branch: 'master',
          label: 'v8.0.0',
          branchLabelMappingKey: '^v8.0.0$',
          isSourceBranch: true,
          mergeCommit: {
            message: 'Change Ulysses to Gretha (conflict) (#8)',
            sha: 'b484e161b705b39dbbfc5005e67ca24d05b23c37',
          },
          number: 8,
          state: 'MERGED',
          url: 'https://github.com/backport-org/backport-e2e/pull/8',
        },
      ]);
    });
  });
});
