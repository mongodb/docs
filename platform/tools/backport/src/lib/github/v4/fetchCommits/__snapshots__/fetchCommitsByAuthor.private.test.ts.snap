// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchCommitsByAuthor snapshot request/response makes the right queries: Query: AuthorId 1`] = `
"query AuthorId($author: String!) {
  user(login: $author) {
    id
  }
}"
`;

exports[`fetchCommitsByAuthor snapshot request/response makes the right queries: Query: CommitsByAuthor 1`] = `
"query CommitsByAuthor($authorId: ID, $commitPath: String, $dateSince: GitTimestamp, $dateUntil: GitTimestamp, $maxNumber: Int!, $repoName: String!, $repoOwner: String!, $sourceBranch: String!) {
  repository(owner: $repoOwner, name: $repoName) {
    ref(qualifiedName: $sourceBranch) {
      target {
        ... on Commit {
          history(
            first: $maxNumber
            author: {id: $authorId}
            path: $commitPath
            since: $dateSince
            until: $dateUntil
          ) {
            edges {
              node {
                ...SourceCommitWithTargetPullRequestFragment
              }
            }
          }
        }
      }
    }
  }
}

fragment SourceCommitWithTargetPullRequestFragment on Commit {
  repository {
    name
    owner {
      login
    }
  }
  sha: oid
  message
  committedDate
  author {
    name
    email
  }
  associatedPullRequests(first: 1) {
    edges {
      node {
        title
        url
        number
        labels(first: 50) {
          nodes {
            name
          }
        }
        baseRefName
        mergeCommit {
          ...RemoteConfigHistoryFragment
          sha: oid
          message
        }
        timelineItems(last: 20, itemTypes: CROSS_REFERENCED_EVENT) {
          edges {
            node {
              ... on CrossReferencedEvent {
                targetPullRequest: source {
                  __typename
                  ... on PullRequest {
                    targetMergeCommit: mergeCommit {
                      sha: oid
                      message
                    }
                    repository {
                      name
                      owner {
                        login
                      }
                    }
                    url
                    title
                    state
                    baseRefName
                    number
                    commits(first: 20) {
                      edges {
                        node {
                          targetCommit: commit {
                            message
                            sha: oid
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}

fragment RemoteConfigHistoryFragment on Commit {
  remoteConfigHistory: history(first: 1, path: ".backportrc.json") {
    edges {
      remoteConfig: node {
        committedDate
        file(path: ".backportrc.json") {
          ... on TreeEntry {
            object {
              ... on Blob {
                text
              }
            }
          }
        }
      }
    }
  }
}"
`;

exports[`fetchCommitsByAuthor snapshot request/response returns the correct response 1`] = `
[
  {
    "author": {
      "email": "soren.louv@elastic.co",
      "name": "Søren Louv-Jansen",
    },
    "sourceBranch": "main",
    "sourceCommit": {
      "branchLabelMapping": {
        "^v(\\d+).(\\d+).\\d+$": "$1.$2",
        "^v8.1.0$": "main",
      },
      "committedDate": "2021-12-20T14:20:16Z",
      "message": "[APM] Add note about synthtrace to APM docs (#121633)",
      "sha": "d421ddcf6157150596581c7885afa3690cec6339",
    },
    "sourcePullRequest": {
      "labels": [
        "Team:APM - DEPRECATED",
        "v8.0.0",
        "release_note:skip",
        "auto-backport",
        "v8.1.0",
      ],
      "mergeCommit": {
        "message": "[APM] Add note about synthtrace to APM docs (#121633)",
        "sha": "d421ddcf6157150596581c7885afa3690cec6339",
      },
      "number": 121633,
      "title": "[APM] Add note about synthtrace to APM docs",
      "url": "https://github.com/elastic/kibana/pull/121633",
    },
    "suggestedTargetBranches": [],
    "targetPullRequestStates": [
      {
        "branch": "8.0",
        "branchLabelMappingKey": "^v(\\d+).(\\d+).\\d+$",
        "isSourceBranch": false,
        "label": "v8.0.0",
        "mergeCommit": {
          "message": "[APM] Add note about synthtrace to APM docs (#121633) (#121643)

Co-authored-by: Søren Louv-Jansen <soren.louv@elastic.co>",
          "sha": "842adfdeb5541b059231857522f9009771a46107",
        },
        "number": 121643,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/121643",
      },
      {
        "branch": "main",
        "branchLabelMappingKey": "^v8.1.0$",
        "isSourceBranch": true,
        "label": "v8.1.0",
        "mergeCommit": {
          "message": "[APM] Add note about synthtrace to APM docs (#121633)",
          "sha": "d421ddcf6157150596581c7885afa3690cec6339",
        },
        "number": 121633,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/121633",
      },
    ],
  },
  {
    "author": {
      "email": "soren.louv@elastic.co",
      "name": "Søren Louv-Jansen",
    },
    "sourceBranch": "main",
    "sourceCommit": {
      "branchLabelMapping": {
        "^v(\\d+).(\\d+).\\d+$": "$1.$2",
        "^v8.1.0$": "main",
      },
      "committedDate": "2021-12-08T20:03:55Z",
      "message": "[APM] Prefer service.name for logs correlation (#120694)",
      "sha": "b1b491959dab47aeb83c88ee2accb2db46d23793",
    },
    "sourcePullRequest": {
      "labels": [
        "release_note:enhancement",
        "Team:APM - DEPRECATED",
        "backport:skip",
        "v8.1.0",
      ],
      "mergeCommit": {
        "message": "[APM] Prefer service.name for logs correlation (#120694)",
        "sha": "b1b491959dab47aeb83c88ee2accb2db46d23793",
      },
      "number": 120694,
      "title": "[APM] Prefer \`service.name\` for logs correlation",
      "url": "https://github.com/elastic/kibana/pull/120694",
    },
    "suggestedTargetBranches": [],
    "targetPullRequestStates": [
      {
        "branch": "main",
        "branchLabelMappingKey": "^v8.1.0$",
        "isSourceBranch": true,
        "label": "v8.1.0",
        "mergeCommit": {
          "message": "[APM] Prefer service.name for logs correlation (#120694)",
          "sha": "b1b491959dab47aeb83c88ee2accb2db46d23793",
        },
        "number": 120694,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/120694",
      },
    ],
  },
  {
    "author": {
      "email": "soren.louv@elastic.co",
      "name": "Søren Louv-Jansen",
    },
    "sourceBranch": "main",
    "sourceCommit": {
      "branchLabelMapping": {
        "^v(\\d+).(\\d+).\\d+$": "$1.$2",
        "^v8.1.0$": "main",
      },
      "committedDate": "2021-12-01T23:26:48Z",
      "message": "[APM] Disable telemetry in agent config endpoint (#120106)",
      "sha": "b1bb4a93959f19a653b9cfb207a5c6acb6559482",
    },
    "sourcePullRequest": {
      "labels": [
        "Team:APM - DEPRECATED",
        "v8.0.0",
        "release_note:skip",
        "auto-backport",
        "v8.1.0",
      ],
      "mergeCommit": {
        "message": "[APM] Disable telemetry in agent config endpoint (#120106)",
        "sha": "b1bb4a93959f19a653b9cfb207a5c6acb6559482",
      },
      "number": 120106,
      "title": "[APM] Disable telemetry in agent config endpoint",
      "url": "https://github.com/elastic/kibana/pull/120106",
    },
    "suggestedTargetBranches": [],
    "targetPullRequestStates": [
      {
        "branch": "8.0",
        "branchLabelMappingKey": "^v(\\d+).(\\d+).\\d+$",
        "isSourceBranch": false,
        "label": "v8.0.0",
        "mergeCommit": {
          "message": "[APM] Disable telemetry in agent config endpoint (#120106) (#120186)

Co-authored-by: Søren Louv-Jansen <soren.louv@elastic.co>",
          "sha": "33c132edf127aafe0886921e7451cbcdb3e68acd",
        },
        "number": 120186,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/120186",
      },
      {
        "branch": "main",
        "branchLabelMappingKey": "^v8.1.0$",
        "isSourceBranch": true,
        "label": "v8.1.0",
        "mergeCommit": {
          "message": "[APM] Disable telemetry in agent config endpoint (#120106)",
          "sha": "b1bb4a93959f19a653b9cfb207a5c6acb6559482",
        },
        "number": 120106,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/120106",
      },
    ],
  },
  {
    "author": {
      "email": "soren.louv@elastic.co",
      "name": "Søren Louv-Jansen",
    },
    "sourceBranch": "main",
    "sourceCommit": {
      "branchLabelMapping": {
        "^v(\\d+).(\\d+).\\d+$": "$1.$2",
        "^v8.1.0$": "main",
      },
      "committedDate": "2021-11-30T18:47:26Z",
      "message": "[APM] Remove index_pattern.json and add custom field formatters (#119915)

* [APM] Remove index_pattern.json and add custom field formatters

* Fix tests

* Fix tests

* Fix tutorial",
      "sha": "434f6e6a88faf24dc1ea41f9f726db78e46355a7",
    },
    "sourcePullRequest": {
      "labels": [
        "Team:APM - DEPRECATED",
        "v8.0.0",
        "release_note:skip",
        "auto-backport",
        "v8.1.0",
      ],
      "mergeCommit": {
        "message": "[APM] Remove index_pattern.json and add custom field formatters (#119915)

* [APM] Remove index_pattern.json and add custom field formatters

* Fix tests

* Fix tests

* Fix tutorial",
        "sha": "434f6e6a88faf24dc1ea41f9f726db78e46355a7",
      },
      "number": 119915,
      "title": "[APM] Remove index_pattern.json and add custom field formatters",
      "url": "https://github.com/elastic/kibana/pull/119915",
    },
    "suggestedTargetBranches": [],
    "targetPullRequestStates": [
      {
        "branch": "8.0",
        "branchLabelMappingKey": "^v(\\d+).(\\d+).\\d+$",
        "isSourceBranch": false,
        "label": "v8.0.0",
        "mergeCommit": {
          "message": "[APM] Remove index_pattern.json and add custom field formatters (#119915) (#120012)

* [APM] Remove index_pattern.json and add custom field formatters

* Fix tests

* Fix tests

* Fix tutorial

Co-authored-by: Søren Louv-Jansen <soren.louv@elastic.co>",
          "sha": "4d5992dbcbfc097f94562894882ad80c2791aad0",
        },
        "number": 120012,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/120012",
      },
      {
        "branch": "main",
        "branchLabelMappingKey": "^v8.1.0$",
        "isSourceBranch": true,
        "label": "v8.1.0",
        "mergeCommit": {
          "message": "[APM] Remove index_pattern.json and add custom field formatters (#119915)

* [APM] Remove index_pattern.json and add custom field formatters

* Fix tests

* Fix tests

* Fix tutorial",
          "sha": "434f6e6a88faf24dc1ea41f9f726db78e46355a7",
        },
        "number": 119915,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/119915",
      },
    ],
  },
  {
    "author": {
      "email": "soren.louv@elastic.co",
      "name": "Søren Louv-Jansen",
    },
    "sourceBranch": "main",
    "sourceCommit": {
      "branchLabelMapping": {
        "^v(\\d+).(\\d+).\\d+$": "$1.$2",
        "^v8.1.0$": "main",
      },
      "committedDate": "2021-11-29T15:12:27Z",
      "message": "[APM] Remove log-log descriptions from correlation charts (#119700)",
      "sha": "ae9d51b7fe3ee6f30d0d196c782e0dcabb7ac5ff",
    },
    "sourcePullRequest": {
      "labels": [
        "Team:APM - DEPRECATED",
        "v8.0.0",
        "release_note:skip",
        "auto-backport",
        "v7.16.0",
      ],
      "mergeCommit": {
        "message": "[APM] Remove log-log descriptions from correlation charts (#119700)",
        "sha": "ae9d51b7fe3ee6f30d0d196c782e0dcabb7ac5ff",
      },
      "number": 119700,
      "title": "[APM] Remove log-log descriptions from correlation charts",
      "url": "https://github.com/elastic/kibana/pull/119700",
    },
    "suggestedTargetBranches": [],
    "targetPullRequestStates": [
      {
        "branch": "8.0",
        "branchLabelMappingKey": "^v(\\d+).(\\d+).\\d+$",
        "isSourceBranch": false,
        "label": "v8.0.0",
        "mergeCommit": {
          "message": "[APM] Remove log-log descriptions from correlation charts (#119700) (#120178)",
          "sha": "0bd9d7f9fe713ad78330fd1aa6c713c9c5c2c8d6",
        },
        "number": 120178,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/120178",
      },
      {
        "branch": "7.16",
        "branchLabelMappingKey": "^v(\\d+).(\\d+).\\d+$",
        "isSourceBranch": false,
        "label": "v7.16.0",
        "mergeCommit": {
          "message": "[APM] Remove log-log descriptions from correlation charts (#119700) (#120179)

# Conflicts:
#	x-pack/plugins/apm/public/components/app/correlations/failed_transactions_correlations.tsx
#	x-pack/plugins/apm/public/components/app/correlations/latency_correlations.tsx
#	x-pack/plugins/apm/public/components/app/transaction_details/distribution/index.tsx",
          "sha": "749d1cfc1c7c33beec8fee58046e3d70dbd43e68",
        },
        "number": 120179,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/120179",
      },
    ],
  },
  {
    "author": {
      "email": "soren.louv@elastic.co",
      "name": "Søren Louv-Jansen",
    },
    "sourceBranch": "main",
    "sourceCommit": {
      "branchLabelMapping": {
        "^v(\\d+).(\\d+).\\d+$": "$1.$2",
        "^v8.1.0$": "main",
      },
      "committedDate": "2021-11-29T13:40:01Z",
      "message": "[APM] Fix bug in documentation on \`span.destination\` metrics (#119789)",
      "sha": "43253ecafa1365eae45d31dd284d0a0bf273bc4d",
    },
    "sourcePullRequest": {
      "labels": [
        "Team:APM - DEPRECATED",
        "v8.0.0",
        "release_note:skip",
        "auto-backport",
        "v8.1.0",
      ],
      "mergeCommit": {
        "message": "[APM] Fix bug in documentation on \`span.destination\` metrics (#119789)",
        "sha": "43253ecafa1365eae45d31dd284d0a0bf273bc4d",
      },
      "number": 119789,
      "title": "[APM] Fix bug in documentation on \`span.destination\` metrics",
      "url": "https://github.com/elastic/kibana/pull/119789",
    },
    "suggestedTargetBranches": [],
    "targetPullRequestStates": [
      {
        "branch": "8.0",
        "branchLabelMappingKey": "^v(\\d+).(\\d+).\\d+$",
        "isSourceBranch": false,
        "label": "v8.0.0",
        "mergeCommit": {
          "message": "[APM] Fix bug in documentation on \`span.destination\` metrics (#119789) (#119840)

Co-authored-by: Søren Louv-Jansen <soren.louv@elastic.co>",
          "sha": "72a88ee6ae55dfacec76cccf0ece7ea6f4bfcc42",
        },
        "number": 119840,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/119840",
      },
      {
        "branch": "main",
        "branchLabelMappingKey": "^v8.1.0$",
        "isSourceBranch": true,
        "label": "v8.1.0",
        "mergeCommit": {
          "message": "[APM] Fix bug in documentation on \`span.destination\` metrics (#119789)",
          "sha": "43253ecafa1365eae45d31dd284d0a0bf273bc4d",
        },
        "number": 119789,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/119789",
      },
    ],
  },
  {
    "author": {
      "email": "soren.louv@elastic.co",
      "name": "Søren Louv-Jansen",
    },
    "sourceBranch": "main",
    "sourceCommit": {
      "branchLabelMapping": {
        "^v(\\d+).(\\d+).\\d+$": "$1.$2",
        "^v8.1.0$": "main",
      },
      "committedDate": "2021-11-16T11:44:55Z",
      "message": "[APM] Remove projections (#118327)

* Delete errors projection

* Remove \`getMetricsProjection\`

* Remove \`getServiceNodesProjection\`

* Fix tests

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>",
      "sha": "b48268dd0c42c9dc1fe310f7c0b3327117d066a5",
    },
    "sourcePullRequest": {
      "labels": [
        "Team:APM - DEPRECATED",
        "v8.0.0",
        "release_note:skip",
        "auto-backport",
        "v8.1.0",
      ],
      "mergeCommit": {
        "message": "[APM] Remove projections (#118327)

* Delete errors projection

* Remove \`getMetricsProjection\`

* Remove \`getServiceNodesProjection\`

* Fix tests

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>",
        "sha": "b48268dd0c42c9dc1fe310f7c0b3327117d066a5",
      },
      "number": 118327,
      "title": "[APM] Remove projections",
      "url": "https://github.com/elastic/kibana/pull/118327",
    },
    "suggestedTargetBranches": [],
    "targetPullRequestStates": [
      {
        "branch": "8.0",
        "branchLabelMappingKey": "^v(\\d+).(\\d+).\\d+$",
        "isSourceBranch": false,
        "label": "v8.0.0",
        "mergeCommit": {
          "message": "[APM] Remove projections (#118327) (#118675)

* Delete errors projection

* Remove \`getMetricsProjection\`

* Remove \`getServiceNodesProjection\`

* Fix tests

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Co-authored-by: Søren Louv-Jansen <soren.louv@elastic.co>",
          "sha": "03657221841ae4c13cc490963e3c2fbcbde0a78e",
        },
        "number": 118675,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/118675",
      },
      {
        "branch": "main",
        "branchLabelMappingKey": "^v8.1.0$",
        "isSourceBranch": true,
        "label": "v8.1.0",
        "mergeCommit": {
          "message": "[APM] Remove projections (#118327)

* Delete errors projection

* Remove \`getMetricsProjection\`

* Remove \`getServiceNodesProjection\`

* Fix tests

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>",
          "sha": "b48268dd0c42c9dc1fe310f7c0b3327117d066a5",
        },
        "number": 118327,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/118327",
      },
    ],
  },
  {
    "author": {
      "email": "soren.louv@elastic.co",
      "name": "Søren Louv-Jansen",
    },
    "sourceBranch": "main",
    "sourceCommit": {
      "branchLabelMapping": {
        "^v(\\d+).(\\d+).\\d+$": "$1.$2",
        "^v8.1.0$": "main",
      },
      "committedDate": "2021-11-15T22:54:40Z",
      "message": "Update query_debugging_in_development_and_production.md (#118491)",
      "sha": "e66377790605274d2610f45eed35c9d8febdb7b9",
    },
    "sourcePullRequest": {
      "labels": [
        "Team:APM - DEPRECATED",
        "v8.0.0",
        "release_note:skip",
        "auto-backport",
        "v8.1.0",
      ],
      "mergeCommit": {
        "message": "Update query_debugging_in_development_and_production.md (#118491)",
        "sha": "e66377790605274d2610f45eed35c9d8febdb7b9",
      },
      "number": 118491,
      "title": "[APM] Add example to docs for \`Inspect ES queries\` setting",
      "url": "https://github.com/elastic/kibana/pull/118491",
    },
    "suggestedTargetBranches": [],
    "targetPullRequestStates": [
      {
        "branch": "8.0",
        "branchLabelMappingKey": "^v(\\d+).(\\d+).\\d+$",
        "isSourceBranch": false,
        "label": "v8.0.0",
        "mergeCommit": {
          "message": "Update query_debugging_in_development_and_production.md (#118491) (#118631)

Co-authored-by: Søren Louv-Jansen <soren.louv@elastic.co>",
          "sha": "81f69039af290230b7b9b0a35924622a031ac572",
        },
        "number": 118631,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/118631",
      },
      {
        "branch": "main",
        "branchLabelMappingKey": "^v8.1.0$",
        "isSourceBranch": true,
        "label": "v8.1.0",
        "mergeCommit": {
          "message": "Update query_debugging_in_development_and_production.md (#118491)",
          "sha": "e66377790605274d2610f45eed35c9d8febdb7b9",
        },
        "number": 118491,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/118491",
      },
    ],
  },
  {
    "author": {
      "email": "soren.louv@elastic.co",
      "name": "Søren Louv-Jansen",
    },
    "sourceBranch": "main",
    "sourceCommit": {
      "branchLabelMapping": {
        "^v(\\d+).(\\d+).\\d+$": "$1.$2",
        "^v8.1.0$": "main",
      },
      "committedDate": "2021-11-08T15:12:52Z",
      "message": "[APM] Ignore unavailable indices for ml jobs (#117632)",
      "sha": "acc7bd9d97a2147f752942dbcf01d7fb2478234b",
    },
    "sourcePullRequest": {
      "labels": [
        "release_note:fix",
        "Team:APM - DEPRECATED",
        "v8.0.0",
        "auto-backport",
        "v8.1.0",
      ],
      "mergeCommit": {
        "message": "[APM] Ignore unavailable indices for ml jobs (#117632)",
        "sha": "acc7bd9d97a2147f752942dbcf01d7fb2478234b",
      },
      "number": 117632,
      "title": "[APM] Ignore unavailable indices for ml jobs",
      "url": "https://github.com/elastic/kibana/pull/117632",
    },
    "suggestedTargetBranches": [],
    "targetPullRequestStates": [
      {
        "branch": "8.0",
        "branchLabelMappingKey": "^v(\\d+).(\\d+).\\d+$",
        "isSourceBranch": false,
        "label": "v8.0.0",
        "mergeCommit": {
          "message": "[APM] Ignore unavailable indices for ml jobs (#117632) (#117872)

Co-authored-by: Søren Louv-Jansen <soren.louv@elastic.co>",
          "sha": "748db6969738ac626be83c80b0d1bed2abd50b70",
        },
        "number": 117872,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/117872",
      },
      {
        "branch": "main",
        "branchLabelMappingKey": "^v8.1.0$",
        "isSourceBranch": true,
        "label": "v8.1.0",
        "mergeCommit": {
          "message": "[APM] Ignore unavailable indices for ml jobs (#117632)",
          "sha": "acc7bd9d97a2147f752942dbcf01d7fb2478234b",
        },
        "number": 117632,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/117632",
      },
    ],
  },
  {
    "author": {
      "email": "soren.louv@elastic.co",
      "name": "Søren Louv-Jansen",
    },
    "sourceBranch": "main",
    "sourceCommit": {
      "branchLabelMapping": {
        "^v(\\d+).(\\d+).\\d+$": "$1.$2",
        "^v8.1.0$": "main",
      },
      "committedDate": "2021-11-08T11:52:20Z",
      "message": "[APM] Document \`_inspect\` flag (#117635)",
      "sha": "d13ee69bdac1deebfea9f02a29cab0f9a87f2f8e",
    },
    "sourcePullRequest": {
      "labels": [
        "Team:APM - DEPRECATED",
        "v8.0.0",
        "release_note:skip",
        "auto-backport",
        "v8.1.0",
      ],
      "mergeCommit": {
        "message": "[APM] Document \`_inspect\` flag (#117635)",
        "sha": "d13ee69bdac1deebfea9f02a29cab0f9a87f2f8e",
      },
      "number": 117635,
      "title": "[APM] Document \`_inspect\` flag",
      "url": "https://github.com/elastic/kibana/pull/117635",
    },
    "suggestedTargetBranches": [],
    "targetPullRequestStates": [
      {
        "branch": "8.0",
        "branchLabelMappingKey": "^v(\\d+).(\\d+).\\d+$",
        "isSourceBranch": false,
        "label": "v8.0.0",
        "mergeCommit": {
          "message": "[APM] Document \`_inspect\` flag (#117635) (#117831)

Co-authored-by: Søren Louv-Jansen <soren.louv@elastic.co>",
          "sha": "276fd505bce0a2ed2163d32c1de70ecf2414ab30",
        },
        "number": 117831,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/117831",
      },
      {
        "branch": "main",
        "branchLabelMappingKey": "^v8.1.0$",
        "isSourceBranch": true,
        "label": "v8.1.0",
        "mergeCommit": {
          "message": "[APM] Document \`_inspect\` flag (#117635)",
          "sha": "d13ee69bdac1deebfea9f02a29cab0f9a87f2f8e",
        },
        "number": 117635,
        "state": "MERGED",
        "url": "https://github.com/elastic/kibana/pull/117635",
      },
    ],
  },
]
`;
